### PR TITLE
Add a basic CSharp section

### LINKED ISSUE #11: Best practices for C#
The best practices for C# are currently missing. They should at least cover the same amount as the current JavaScript ones.

### PR DESCRIPTION
**Pull request by bkaestner**
As there is no CSharp section in Codewars at the moment, there should at least be a basic one in this document. It should contain the following features:
- [x] links to NUnit
- common misconceptions
  - [x] wrong order of arguments
  - [x] bad test names
  - [x] ???
- slightly advanced tricks
  - [x] `[TestCase]`
  - [x] `[Random]`
  - [x] `[Theory]`

Merging this pull-request would also fix #11.

### CODE REVIEW

File: rules/0230-CSharp.md [hunk 1]
```diff
@@ -1 +1,121 @@
...
+
+### Use descriptive test names
+This was already addressed in the previous section, but it's rather important
+in CSharp. NUnit doesn't give you any method to label a test. Instead, a test
+is simply a public static method with a `[Test]` attribute inside a class with
+the `[TestFixture]` attribute. The _name_ of your test is the method's name.
+
+Make that name descriptive. A method called `test1` doesn't help.
+
+### Use `[TestCase]` for multiple test cases
```
Comment (jhoffner) [inline rules/0230-CSharp.md]: I never knew this!

**Comment by jhoffner** [comment]
This looks really good to me :+1:

File: rules/0230-CSharp.md (hunk 1 above)
Comment (bkaestner) [inline rules/0230-CSharp.md]: I only came across `[TestCase]` by chance. @Killfile posted [a comment](http://www.codewars.com/kata/559590633066759614000063/discuss#55c505765d0910da98000120) and @g964's comment made me have a look at NUnit's documentation.

**Comment by bkaestner** [comment]
Thanks. I guess I check GitHub's suggested maximum line length, spelling and style in the evening and then merge, unless someone finds errors or mandatory missing content.

**Pull request merged**